Add procedure to troubleshoot permission issues (#3271)
theforeman/foreman-documentation: guides/common/modules/proc_troubleshooting-permission-issues.adoc

[id="troubleshooting-permission-issues"]
= Troubleshooting permission issues

> You can perform a health check on {ProjectServer} to find permission issues.

Reviewer: "Permission issues" can mean a lot of things, not just fixing duplicate permissions. So using "permission issues" in the heading and intro here is too vague and can be misleading. Can you consider a more precise description of what the section contains?

Reviewer: The revised introduction along with the move to Upgrading provide sufficient context. You can resolve this thread.

> You can perform a health check on {ProjectServer} to find permission issues.

Reviewer: Can you include a bit more context into the introduction? How will users know that they are in fact dealing with permission issues? Is there a specific action failure, an error message, etc.? Or something that  you expect users would try to look up with a full text search?

Author: I have slightly extended the introduction and referenced the redmine issue in my commit message. Should I include parts of the error message?

Reviewer: I like the new introduction, thanks.

Regarding the error message: If the message is something you expect users might first see in their environment and then search for what it means, then yes, including it is a good idea. If you think this scenario is unlikely, then including the message is not necessary. Ultimately, your call :)

.Procedure

> . On your {ProjectServer}, find duplicate permissions:

Reviewer: Lines 7 and 21 seem to inconsistently describe what the command does. On line 7, you write that the purpose of the command is to "find duplicate permissions"; on line 21, it's "missing or duplicate permissions". Can you try to describe the command in both places the same way?

+
[options="nowrap", subs="verbatim,quotes,attributes"]
----
# {foreman-maintain} health check --label duplicate_permissions
----
. Fix duplicate permission issues:
+
[options="nowrap", subs="verbatim,quotes,attributes"]
----
# foreman-rake db:seed
----

.Verification

> * Rerun the check to ensure that there are no missing or duplicate permissions:

Reviewer: ```suggestion
* Rerun the check to ensure no missing or duplicate permissions remain:
```
"There is/are" is considered an empty phrase.

[id="troubleshooting-permission-issues"]
= Troubleshooting permission issues

{Project} upgrades perform pre-upgrade checks.

> If your {Project} upgrade fails due to permission issues, you can identify and remedy them on your {ProjectServer}.
> 
> You can perform this procedure if you receive an error similar to the following one:
> 
> [options="nowrap", subs="verbatim,quotes,attributes"]
> ----
> 2024-01-29T20:50:09 [W|app|] Could not create role 'Ansible Roles Manager': ERF73-0602 [Foreman::PermissionMissingException]: some permissions were not found:
> ----

Reviewer: I'd like to find a way to avoid self-referential wording ("This procedure").

Also, it's not the upgrade that fails, is it? I'd like to assume the error is displayed in the pre-upgrade check phase, before the upgrade itself even starts.

I tried shuffling words around a bit and would suggest something like this. What do you think?

```suggestion
If the pre-upgrade check discovers permission issues, it fails with an error similar to the following one:

[options="nowrap", subs="verbatim,quotes,attributes"]
----
2024-01-29T20:50:09 [W|app|] Could not create role 'Ansible Roles Manager': ERF73-0602 [Foreman::PermissionMissingException]: some permissions were not found:
----

If you see an error like this on your {ProjectServer}, identify and remedy the permission issues.
```

Author: Yes, only the pre-upgrade check fails, but that means as a user, the upgrade process fails.

Applied.
